### DISCUSSION TITLE
Garmin Connect activity title

### DISCUSSION THREAD

**Discussion by jochen35** [philosowaffle/peloton-to-garmin#562] [General]
Hello, is there a way to customize the title when uploading to Garmin Connect? Either so that with Garmin Connect only “Peloton Ride” is preceded by at least the addition “Peloton -”. So you can recognize Peloton activities as such at first glance. Greetings Jochen

**Comment by philosowaffle** [comment]
Not currently but sounds like a good idea. I created feature request #564 for this.

**Comment by geoffbon** [comment]
I came to ask almost the same thing as the OP, except that I was going to suggest allowing a token-based approach.  I want to name my workouts in Garmin something like 
Peloton {class name}, {instructor name}
E.g. "Peloton 15 min HIIT Ride, Ben Alldis"
So I'm picturing tokens for class name, instructor, length, activity type, etc.

It sounds like you maybe did the prefix, is it feasible to consider the option to use the other pieces of information?  

I did look through the code, and I'm still getting my head around it - there's a lot, but well laid out!  I'm not confident enough yet to supply a pull request, but it looks like there's enough info to provide this from a data point of view?

**Reply by philosowaffle** [reply]
That's a good idea.  I've made a new feature request for this. #591

**Reply by geoffbon** [reply]
The Handlebars template option looks really good!  
I was able to set up the title how I wanted to, but it doesn't seem to transfer through at all.

Here's what I have in the title:
![image](https://github.com/philosowaffle/peloton-to-garmin/assets/54509333/36d3aa4f-f845-46e4-822c-fb0bd991adef)

Here's what I see in Garmin Connect:
![image](https://github.com/philosowaffle/peloton-to-garmin/assets/54509333/d982ca5b-8cc2-4d44-8f7a-b150159811e6)

Am I missing a setting somewhere?

Edit: I'm on 4.1.0, converting to FIT, uploading FIT to Garmin.

**Reply by philosowaffle** [reply]
hmmm that's unexpected. I'll have to poke at it

#608

**Reply by geoffbon** [reply]
Cheers - let me know if I can help with any info / logs etc.

**Reply by philosowaffle** [reply]
Could you send me what the name of the Peloton class is?  Wasn't able to repro on my own so now I'm wondering if there is a special character in the class name itself that isn't being handled gracefully.

**Reply by geoffbon** [reply]
Sure- the class was `30 min EDM Ride` with `Hannah Frankson`
The log says 
`Uploading to Garmin: 
C:\temp\PelotonToGarmin\working\upload\75ca0d67a2854e9b8763d06922fcdd20_Peloton_30_min_EDM_Ride,_Hannah_Frankson.fit`

...but that 'working' directory doesn't exist.  Does it get created and destroyed every time?
Also, perhaps the comma in my template is throwing something off?

**Reply by philosowaffle** [reply]
hmmm yeah it must be the comma.  I tested in Garmin Connect earlier today and using comma's in activity titles was perfectly fine, but I admit I did not end-to-end test if it worked when P2G sent the title with a comma.  I can test that tmrw.

And yes, that's correct, the working directory is temporary, gets created/destroyed as needed.

**Reply by geoffbon** [reply]
Thanks, will be interested to hear how the test goes at your end, it feels like something else might be off.  I can test it too of course, without the comma - would be weird for the API to forbid it, but the online Connect UI to allow it right?  Because I can easily enough go in and change it to include a comma after the upload.

**Reply by geoffbon** [reply]
Can confirm it's not the comma, changing it to "with" results in the same "Indoor Cycling" activity on Garmin Connect.

**Reply by geoffbon** [reply]
One more note on this... I did always get that same "Indoor Cycling" title even before the switch to templating; is that what everyone else gets?

**Reply by philosowaffle** [reply]
Oh now that's interesting. No, the title "Indoor Cycling" is not what P2G historically used.  P2G historically set the title to `Class Name with Instructor Name`, similar to the default Template added in the settings.

On Garmin's end I know there are some settings for how to name activity files, I wonder if one of those choices doesn't play nicely with P2G?  (just brainstorming)

(sorry for late response, I'm still dealing with near constant internet outages in my area... going on day four now)

**Reply by geoffbon** [reply]
Is it possibly to do with this setting?  Garmin Connect on the web > Profile > Profile and Account > Account Settings.
Having it set this way names outdoor runs with the place name, which I like:

![image](https://github.com/philosowaffle/peloton-to-garmin/assets/54509333/0ea4c0c9-a68d-4956-90c0-c0e49356bc59)

I just tried syncing a workout with the next setting down (Location and Device Activity Name), same result.  I can try the other settings as I have more workouts to use.  What do you have yours set to?

**Reply by philosowaffle** [reply]
This might be the culprit. I have mine set to `Workout Name (when available)`.  From what I can tell in my own Activities, Garmin falls back to using Location & Type when no workout name is found. For example, all of my runs recorded on my watch say "[Location} Running"

Having trouble finding Garmin docs about this setting.

**Reply by geoffbon** [reply]
Ok great, will try changing to your setting; I'm running tonight, and doing a Peloton spin tomorrow, so that should test both.  If this turns out to be the answer, could perhaps be added to docs somewhere?

**Reply by geoffbon** [reply]
Holy fricken guacamole, that was it - thanks!  After changing that setting to `Workout Name` it now syncs with the custom name for the Peloton (which goes through P2G), and the same name it always did for the run (which syncs directly from the watch), e.g. `Miami Running`.
You should be able to close that issue now, and let me know if I can help with a PR for the docs (edit: done).
